### PR TITLE
[GPII-3718] Move CouchDB alerts to the CouchDB module

### PR DESCRIPTION
**Pull request by amatas**
This PR moves the CouchDB related alerts to the CouchDB Module.

### CODE REVIEW

File: gcp/modules/gcp-stackdriver-monitoring/alert_dns_modify.tf [hunk 1]
```diff
@@ -31,5 +33,31 @@ resource "google_monitoring_alert_policy" "dns_modify" {
...
+      while [ "$ALERT_READY" != 'true' ] && [ "$COUNT" -le "$MAX_RETRIES" ]; do
+        echo "Waiting for log based metric dns.modify to be ready ($COUNT/$MAX_RETRIES)"
+        gcloud logging metrics describe dns.modify > /dev/null
+        [ "$?" -eq 0 ] && ALERT_READY=true
+        # Sleep only if we're not ready
+        [ "$ALERT_READY" != 'true' ] && sleep "$SLEEP_SEC"
+        COUNT=$((COUNT+1))
+      done
+    EOF
+  }
```
Comment (stepanstipl) [inline gcp/modules/gcp-stackdriver-monitoring/alert_dns_modify.tf]: Are these changes related to CouchDB alerts in any way? If not can we leave them for a separate PR?

**Comment by amatas** [comment]
> I've noticed one issue, that might help to resolve the dependency race condition between LBM metrics and alerts, without having to use the `null_resource` wait "hack".
> 
> Currently, the alert policy does not reference the metric in any way, i.e. metric name `compute.disks.createSnapshot` is hardcoded, so TF is not aware of the relationship. Referencing the name in the alert policy via a variable - `${google_logging_metric.disks_createsnapshot.name}` - would make the relationship visible to TF and might help to prevent the observed issue.
> 
> (similar for other alerts/metrics)

The null_resource has the depends_on parameter, and the alert depends on the null_resource. Before this change, the alert had the dependency of the lbm https://github.com/gpii-ops/gpii-infra/pull/566/commits/516da38b6ef98bdba2b26ca5615294da29e9b4fe#diff-db551adaf13904e858b436155c52bb3bR6

File: gcp/modules/gcp-stackdriver-monitoring/alert_dns_modify.tf (hunk 1 above)
Comment (amatas) [reply gcp/modules/gcp-stackdriver-monitoring/alert_dns_modify.tf]: It is not related to the CouchDB but it is one of the most raised errors when I tested the deployment using my cluster. Here it is a failure in the pipeline related:
https://gitlab.com/gpii-ops/gpii-infra/-/jobs/401492401

**Comment by stepanstipl** [comment]
> The null_resource has the depends_on parameter, and the alert depends on the null_resource. Before this change, the alert had the dependency of the lbm 516da38#diff-db551adaf13904e858b436155c52bb3bR6

Ok, thanks @amatas, somehow it's not showing in the PR diff.


One more question - how is this PR solving the problem? Resp. how is it different to what Terraform provider already seems to be doing (reading metric after creating it - https://github.com/terraform-providers/terraform-provider-google/blob/8ee9fef853489afc60888de21c80205ea04f7aa0/google/resource_logging_metric.go#L348)? And has it been tested, how?

Have you considered any alternatives, such as fixing in the TF provider? And is the same issue still happening with the latest `v3.4.0` version of the provider?

**Comment by amatas** [comment]
I have removed the wait script as it seems that does not make difference with or without it. As this PR is about moving CouchDB alerts to the CouchDB module I would like to not make more changes in this PR, further changes will be addressed in separates PRs.

**Comment by natarajaya** [comment]
@amatas Nice refactoring, LGTM!
